test/e2e/tests/change-language.spec.js

```diff
@@ -0,0 +1,325 @@
+  ensInput: '[data-testid="ens-input"]',
+  nftsTab: '[data-testid="home__nfts-tab"]',
+};
+
+async function changeLanguage({ driver, languageIndex }) {
+  await driver.clickElement(selectors.accountOptionsMenuButton);
+  await driver.clickElement(selectors.settingsOption);
+
+  const dropdownElement = await driver.findElement(selectors.localeSelect);
+  await dropdownElement.click();
+
+  const options = await dropdownElement.findElements(By.tagName('option'));
+  await options[languageIndex].click();
+}
+
+describe('Settings - general validate the change language functionality', function () {
+  it('User selects "Español" language and verify that changing the language from the default', async function () {
+    const languageIndex = 10;
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+        failOnConsoleError: false,
+      },
+
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        await changeLanguage({ driver, languageIndex });
+        // Validate the label changes
+        const advanceText = await driver.findElement({
+          tag: 'span',
+          text: 'Idioma actual',
+        });
+        assert.equal(
+          await advanceText.getText(),
+          'Idioma Actual',
+          'Language did not change',
+        );
+      },
+    );
+  });
+
+  it('User selects "Español" language from the dropdown and change back to english and verify that the word is correctly changed back to english', async function () {
+    let languageIndex = 10;
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+        failOnConsoleError: false,
+      },
+
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await changeLanguage({ driver, languageIndex });
+        const advanceText = await driver.findElement({
+          tag: 'span',
+          text: 'Idioma actual',
+        });
+        assert.equal(
+          await advanceText.getText(),
+          'Idioma Actual',
+          'Language did not change',
+        );
+
+        await driver.delay(2000);
+
+        languageIndex = 9;
+        const dropdownElement = await driver.findElement(
+          selectors.localeSelect,
+        );
+        await dropdownElement.click();
+        console.log(' dropdownElement:', dropdownElement);
+        const options = await dropdownElement.findElements(
+          By.tagName('option'),
+        );
+        await options[languageIndex].click();
+
+        await driver.delay(2000);
+
+        const labelText = await driver.findElement(
+          By.css('.settings-page__content-label'),
+        );
+        assert.equal(
+          await labelText.getText(),
+          'Current Language',
+          'Language did not change',
+        );
```

Comment:
This is an example of the "find then assert" race condition
https://consensys.slack.com/archives/GTQAGKY5V/p1686238528522549

Reply:
I made the code changes and marking resolved.